### ISSUE TITLE
Long Return Time after Exitgng External Program

### ISSUE THREAD

**Issue by twizz0r** [ModOrganizer2/modorganizer#643]
## The problem:
After exiting an external program like xEdit, MatorSmash, Merge Plugins or Wrye Flash, MO2 repeatedly tries to close plugins.txt until it succeeds causing 45+ second waits for return to functionality. 

## To Reproduce:

1. Run xEdit through MO2 and exit. 

## Environment:

* Mod Organizer Version that exhibits the issue: 2.2.0alpah10 with updates
* Last Mod Organizer Version that did not exhibit the issue (if applicable): Mod.Organizer-2.1.7alpha6
* Desktop OS/version used to run Mod Organizer: Win 8.1 Pro

## Details:
If necessary, describe the problem you have been experiencing with more details and pictures.

### USVFS:
https://gist.github.com/twizz0r/e1a9e205d531e5ada1afbd85f3893efe#file-usvfs-2019-02-01_19-32-37-log

### MO Interface:
https://gist.github.com/twizz0r/fe75ffeb15044ecbefa341b29ac2b2e5#file-usvfs-2019-02-01_20-07-19-log

**Comment by DefinitelyNotFred** [comment]
I'm having the same issue, so I'll dump all I know here. I first mentioned this on the Testers channel on the Discord (On Wednesday at 1:30 PM) but haven't been online recently to make any replies so I decided to come here and restart a topic and in the process found twizz0r's topic.

My wait times are closer to the 25 seconds mark. El Rizzo on Discord suspected it might be something to do with archive parsing, but I've checked and don't have that enabled. AL asked if MO2 stays locked in the time period I'm talking about. Well, the screenshot below is what it looks like for the entire time when it hangs up for me. Clicking things won't give me any response but hovering over plugins with the mouse will highlight them and bring up the tooltip like usual. Maybe that green loading bar at the bottom can give you a bit of a hint on where to start looking?

And to be clear, since it seemed like there was a bit of confusion about this on the Discord, this seems to happen for _any_ executable run through MO2, not just FNVEdit.

![mo2 executable hangup](https://user-images.githubusercontent.com/30205205/52172102-50dab480-272e-11e9-80ba-b5aa43d949be.png)

I'm currently using MO2 2.2.alpha9
Operating system is Windows 10 Home

**Comment by Al12rs** [comment]
The fact that you are both modding FNV and there haven't been any other reports on this makes it pretty clear this must be an issue with the FNV Mo2 plugin.

Thank you both very much for reporting this with detailed info!

**Comment by twizz0r** [comment]
YW! 

I'm glad DefinitelyNotFred put up that screen shot. Notice the time jump backward in the log windows on closing plugins.txt. Same for me. I just couldn't capture it. 

I also have archive parsing disabled.

**Comment by Al12rs** [comment]
Can you guys confirm whether the plugin loadorder is compromised or not after running the external executable?
I have a report with similar logs for Oldrim but the issue there seems to be that load order gets set to alphabetical instead of longer time to unlock.

**Comment by twizz0r** [comment]
Not happening here. LO stays the same. Wrye Flash *can* mess up LOs when it does its own redating but that's something that can be fixed inside of Wrye.

**Comment by DefinitelyNotFred** [comment]
Mildly interesting fact: opening LOOT through MO2, pressing unlock, closing LOOT, and then refreshing the mod list in MO2 will cause two long wait periods: once after pressing unlock and second after refreshing the mod list. This happens even if nothing is actually rearranged by LOOT. When doing the same using xEdit or MergePlugins, it only seems to have that waiting period upon unlocking MO2 and not upon refreshing (but maybe that's just an issue of them being used in a way that doesn't touch as many plugins at once.)

**Comment by twizz0r** [comment]
Fun fact: external programs includes the game itself ;)

**Comment by Al12rs** [comment]
yeah this needs some looking into asap. I won't have time for this until after the 15th though

**Comment by twizz0r** [comment]
No worries. Everything still works fine.

**Comment by Al12rs** [comment]
This seems to be limited to games that use filetime to determine loadorder

**Comment by Al12rs** [comment]
I have investigated and this seems to be caused by the new archive conflicts code, I put an extra condition to avoid completely the archive loading part instead of simply passing empty archives to the function.

That does not solve the issue that for older games getting the load order is much slower since it has to read the filetime from the files on disk, making archive conflicts incredibly slow when refreshing the modlist.
This is because the plugins loadorder is parsed twice for each mod during a full refresh, to get  the correct file priority of the files inside archives.

My change should fix the issue while not using archive conflicts.

**Comment by twizz0r** [comment]
Looks like this is fixed in the new build. tyvm!

**Comment by Al12rs** [comment]
It's fixed only if you don't use archive conflicts. We need to find a way to improve refresh performance there, mostly avoid having to get load order for each mod added

**Comment by DefinitelyNotFred** [comment]
The new build fixed it for me as well.
